whois-api/src/test/java/net/ripe/db/whois/api/rest/SyncUpdatesServiceTestIntegration.java

```diff
@@ -453,6 +455,47 @@ public void create_person_with_changed_attributes() {
         assertThat(databaseHelper.lookupObject(ObjectType.PERSON, "TP2-RIPE").containsAttribute(AttributeType.CHANGED), is(false));
     }
 
+    @Test
+    public void modify_generated_attributes_changes_last_modified_attribute() {
+        databaseHelper.insertUser(User.createWithPlainTextPassword("agoston", "zoh", ObjectType.AUT_NUM));
+        final ZonedDateTime oldDateTime = testDateTimeProvider.getCurrentDateTimeUtc();
+        testDateTimeProvider.setTime(oldDateTime.toLocalDateTime());
+
+        final String AUTNUM_TEST = "" +
+                "aut-num:        AS104\n" +
+                "status:         ASSIGNED\n" +
+                "as-name:        End-User-2\n" +
+                "admin-c:        TP1-TEST\n" +
+                "tech-c:         TP1-TEST\n" +
+                "mnt-by:         mntner-mnt\n" +
+                "created:         2001-02-04T17:00:00Z\n" +
+                "last-modified:   2001-02-04T17:00:00Z\n" +
+                "source:         TEST\n";
+
+        databaseHelper.addObject(PERSON_ANY1_TEST);
+        databaseHelper.addObject(MNTNER_TEST_MNTNER);
+        databaseHelper.addObject(AUTNUM_TEST);
+
+        final CIString orginialModifiedDate = databaseHelper.lookupObject(ObjectType.AUT_NUM, "AS104").getValueForAttribute(AttributeType.LAST_MODIFIED);
+
+        final String response = RestTest.target(getPort(), "whois/syncupdates/test?" +
+                "DATA=" + SyncUpdateUtils.encode("aut-num:        AS104\n" +
+                "status:         OTHER\n" +
+                "as-name:        End-User-2\n" +
+                "admin-c:        TP1-TEST\n" +
+                "tech-c:         TP1-TEST\n" +
+                "mnt-by:         mntner-mnt\n" +
+                "source:         TEST-NONAUTH\n" +
+                "override:       agoston,zoh\n"))
+                .request()
+                .get(String.class);
+
+        assertThat(response, containsString("Modify SUCCEEDED: [aut-num] AS104"));
+        assertThat(databaseHelper.lookupObject(ObjectType.AUT_NUM, "AS104").getValueForAttribute(AttributeType.LAST_MODIFIED), is(not(orginialModifiedDate)));
```

Comment:
If tests are using 'testDateTimeProvider' then time doesn't move forwards between tests, so last-modified will be the same, always?

Reply:
Actually when creating a object, I specifically added last-modified to  " 2001-02-04T17:00:00Z". So while adding it is not using the testDateTimeProvider. 

Also assertion is done on comparing the last-modified before and after the update and it should be not equal.
But yes for readability I will move testDateTimeProvider time between creating the aut-num and updating the aut-num

Reply:
Done

Reply:
Thanks Mahesh!